android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js

```diff
@@ -7,21 +7,12 @@
 
 browser.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
   switch (message.action) {
-     case 'show':
-       let readerViewUrl = new URL(browser.runtime.getURL("/readerview.html"));
-       readerViewUrl.searchParams.append("id", sender.contextId);
-       readerViewUrl.searchParams.append("url", message.url);
-       readerViewUrl.searchParams.append("colorScheme", message.options.colorScheme);
-       browser.tabs.update({url: readerViewUrl.href}).catch((e) => {
-           console.error("Failed to open reader view", e, e.stack);
-       });
-       break;
      case 'addSerializedDoc':
-        storage.session.set(sender.contextId.toString(), message.doc);
```

Comment:
Ahah this is good context, but the reason for the change is actually quite simpler. We use the `contextId` as a the cacheId on this line and then it sets it as the [`id` URL parameter](https://searchfox.org/mozilla-mobile/rev/47446b701ada1175292df428f429a69091b3caf1/firefox-android/android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview-background.js#12) when redirecting to `readerview.html`. The readerview script than takes that [id](https://searchfox.org/mozilla-mobile/rev/47446b701ada1175292df428f429a69091b3caf1/firefox-android/android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview.js#305) and [passes it](https://searchfox.org/mozilla-mobile/rev/47446b701ada1175292df428f429a69091b3caf1/firefox-android/android-components/components/feature/readerview/src/main/assets/extensions/readerview/readerview.js#286) back to the background script to get the cached document.

Since we're now handling this flow in the `ReaderViewFeature` 

> nit: we can reduce round-trip time in a review if we can be more descriptive on what this change is doing and why.
This is a very good point. I will do a better job at documenting all the changes in more complex patches in the future.